tests/suites/test_suite_constant_time_hmac.function

```diff
@@ -81,8 +81,19 @@ void ssl_cf_hmac(int hash)
      * Test all possible lengths up to a point. The difference between
      * max_in_len and min_in_len is at most 255, and make sure they both vary
      * by at least one block size.
+     *
+     * Under Valgrind, test a subset of lengths to save on CI time.
      */
     for (max_in_len = 0; max_in_len <= 255 + block_size; max_in_len++) {
+#if defined(MBEDTLS_VALGRIND)
```

Comment:
I really don't like having code that sliently does different things depending on the “test intensity”. Why would we want to test less with Valgrind than without? What makes this particular subset representative enough, but not representative enough when not using Valgrind?

If it does make sense to make a difference here, we should have separate test cases, some of which are explicitly skipped.

Reply:
You only get instruction coverage from lcov, not relevant-case coverage, so it isn't very informative. IIRC that test is particularly thorough because the first iteration of the library function had an off-by-one error that caused it to have the wrong result 1/256 of the time.

Reply:
That seems like over-fitting to the last bug - if we checked for off-by-one errors in this manner for every part of the code, the tests would take months to run. And there's no particular reason for this piece of code to be more likely to have off-by-one errors now. So I think it's right to dial back the testing here.

@mpg introduced this, I think - WDYT Manuel?

Reply:
> That seems like over-fitting to the last bug

To the last few bugs actually. We had a number of off-by-one our "anti Lucky 13" code over the years. It is significantly more tricky than the rest of our code, given you have to reason about the padding having any length between 1 and max(256, total length). So when I last revised this part of the code, I wanted to be really sure.

Now I agree that the test is slow to the point of being annoying, and I can't say for sure we need to test absolutely all those values. Perhaps a handful of well-chosen values around critical thresholds would do just as well. But as Gilles says, lcov is of no help here.